stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Channel.kt

```diff
@@ -0,0 +1,69 @@
+package io.getstream.chat.android.ui.utils.extensions
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import androidx.core.content.ContextCompat
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.ui.R
+import java.util.Date
+
+internal fun Channel.getUsers(excludeCurrentUser: Boolean = true): List<User> =
+    members
+        .map { it.user }
+        .let { users ->
+            when {
+                excludeCurrentUser -> users.withoutCurrentUser()
+                else -> users
+            }
+        }
+
+internal fun Channel.getDisplayName(): String = name.takeIf { it.isNotEmpty() }
+    ?: getUsers().joinToString { it.name }
+
+internal fun Channel.getLastMessage(): Message? =
+    messages
+        .filter { it.createdAt != null || it.createdLocallyAt != null }
+        .filter { it.deletedAt == null }
+        .filter { it.type == "regular" } // migrate ModelType?
+        .maxByOrNull { it.createdAt ?: it.createdLocallyAt!! }
+
+internal fun Channel.getLastMessageDisplayText(): String =
+    getLastMessage()?.let { "${it.getDisplayName()}: ${it.text}" } ?: String.EMPTY
```

Comment:
Looks convenient, but I don't think we should do string concatenation for the UI in extension functions

Reply:
Would you rather it be a string template with placeholders in `strings.xml`?